### ISSUE TITLE
Please move username and password data into separate file

### ISSUE THREAD

**Issue by ALIENQuake** [znck/grammarly#3]
So users can exclude it when syncing VSCode user configuration with Github.

**Comment by znck** [comment]
Is there a known way to do this?

**Comment by ALIENQuake** [comment]
@znck 
You can simply add `grammarly.privatesettings` configuration item with default to `FolderWithVSCodeUserSettings\settings-grammarly.json` then read and write username and password there instead of VSCode main settings.json.

**Comment by mwz** [comment]
@ALIENQuake, I can't get this to work.
Are you saying that in my vscode settings `settings.json` file I should add

```json
"grammarly.privatesettings": "~/Library/Application Support/Code/User/settings-grammarly.json"
```
and extract all the grammarly settings there, like so?

```json
{
  "grammarly.dialect": "british",
  "grammarly.domain": "technical",
  "grammarly.password": "redacted",
  "grammarly.username": "redacted",
  "grammarly.userWords": []
}
```

The settings key `grammarly.privatesettings` doesn't seem to be recognised as a valid key and the extension doesn't seem to be connecting to my account, i.e. I can't add any custom words to my user dictionary.

**Comment by ALIENQuake** [comment]
@mwz The issue is not closed so I assume that it isn't yet implemented unless I miss something.

**Comment by znck** [comment]
It's not implemented yet.

**Comment by andrew-codes** [comment]
Would saving the settings using the [global storage path](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext.globalStoragePath) be the correct approach? I think it provides a directory for each extension; with both read and write access.

**Comment by ALIENQuake** [comment]
@andrew-codes Sure, everything is better than user config.

**Comment by andrew-codes** [comment]
I'm going to take a stab at this later today and submit a PR. I, too, sync my settings to Gists. Storing the credentials in settings is a blocker for me also. From a design perspective and desired patterns for this code base, are there any considerations I should take @znck ?

**Comment by znck** [comment]
I was planning to store cookie in some place and prompt user for credentials. By storing cookie, it would reuse session that mean same container and document namespace hence documents preserve manuals fixes even after vscode restart (just a hypothesis, need to test). 

As per conventions, just use prettier.

**Comment by andrew-codes** [comment]
Will storing it in a cookie enable the same credentials across different instances and workspaces of VS Code, too? It does appear to be beneficial to use the same container/namespace if it preserves manual fixes to documents across restarts. Are you thinking of storing the credentials in the cookie or the output of using the credentials when authenticating?

I'll try the cookie approach first and test the hypothesis and resort to a settings file in the global storage path as a last-ditch option. I wonder what best practices are out there for storing sensitive information for VSCode extensions. It may be a good idea to look into that as well.

**Comment by znck** [comment]
In current auth process, it collects cookie header on successful authentication and use the cookie in subsequent requests. 

We still have to use a global store to store the cookie.

**Comment by znck** [comment]
Check `grammarly-auth.ts` for auth related code.

**Comment by andrew-codes** [comment]
Ah ok. I misunderstood. I thought you wanted to store the username/password in a cookie within VS Code, so I wasn't following that train of thought. However, is this a correct summary? Store the username/password AND any relevant cookie data to a separate place/file instead of storing (only) username/password in VS Code's settings.

If the above is correct, then I think I understand your intentions and can open a PR for it (hopefully later today or tomorrow).

**Comment by ALIENQuake** [comment]
@andrew-codes Yes, that's the correct intention, at the very least from the perspective of my request.

**Comment by znck** [comment]
👍 @andrew-codes Please go ahead with the PR

**Comment by andrew-codes** [comment]
I did a little research and found a recommended way to store secrets like username, passwords, tokens, etc. It is to use keytar that already ships with VS Code. I adapted some code from the extension vscode-pull-request-github ([src](https://github.com/microsoft/vscode-pull-request-github/blob/master/src/authentication/keychain.ts)).

However, I am beginning to run into a strange issue. I am now experiencing this issue when using your master branch (including blowing away and re-installing node_modules). I'll mention the problem in the PR I open, and maybe we can determine if this is my environment only, the code, or some mixture of the two.
